### PR TITLE
(#5224)some little fix for spark load

### LINKED ISSUE #5224: [BUG] StringType can not transform to DateType when read hive table to doris by spark load
**Describe the bug**
I test doris global dictionary function by spark load.  The partition field in hive table is String type and the corresponding field in doris table is DATE type. Then I submitted a spark load job, but the job failed and I got the error message.

```
spark etl job run failed
org.apache.spark.SparkException: Job aborted due to stage failure: Task 15 in stage 13.0 failed 4 times, most recent failure: Lost task 15.3 in stage 13.0 (TID 1370, plat-data-hadoop-pro334, executor 21): java.lang.RuntimeException: Error while encoding: java.lang.RuntimeException: java.lang.String is not a valid external type for schema of date
staticinvoke(class org.apache.spark.sql.catalyst.util.DateTimeUtils$, DateType, fromJavaDate, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 0, logdate), DateType), true, false) AS logdate#240
validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 1, app_id), IntegerType) AS app_id#241
staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 2, game), StringType), true, false) AS game#242
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 3, event_action), StringType), true, false) AS event_action#243
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 4, event_category), StringType), true, false) AS event_category#244
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 5, event_label), StringType), true, false) AS event_label#245
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 6, event_value), StringType), true, false) AS event_value#246
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 7, game_level), IntegerType) AS game_level#247
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 8, game_vip_level), IntegerType) AS game_vip_level#248
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 9, game_tag), IntegerType) AS game_tag#249
if (assertnotnull(input[0, org.apache.spark.sql.Row, true]).isNullAt) null else validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 10, event_ts), LongType) AS event_ts#250L
staticinvoke(class org.apache.spark.unsafe.types.UTF8String, StringType, fromString, validateexternaltype(getexternalrowfield(assertnotnull(input[0, org.apache.spark.sql.Row, true]), 11, uuid), StringType), true, false) AS uuid#251
	at org.apache.spark.sql.catalyst.encoders.ExpressionEncoder.toRow(ExpressionEncoder.scala:292)
	at org.apache.spark.sql.SparkSession$$anonfun$4.apply(SparkSession.scala:593)
	at org.apache.spark.sql.SparkSession$$anonfun$4.apply(SparkSession.scala:593)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:435)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:441)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at org.apache.spark.util.collection.ExternalSorter.insertAll(ExternalSorter.scala:191)
	at org.apache.spark.shuffle.sort.SortShuffleWriter.write(SortShuffleWriter.scala:62)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: java.lang.String is not a valid external type for schema of date
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.StaticInvoke_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.writeFields_0_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificUnsafeProjection.apply(Unknown Source)
	at org.apache.spark.sql.catalyst.encoders.ExpressionEncoder.toRow(ExpressionEncoder.scala:289)
	... 20 more

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1889)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1877)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1876)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1876)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:926)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2110)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2059)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2048)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:737)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2061)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2082)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2101)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:935)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:933)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:363)
	at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:933)
	at org.apache.spark.api.java.JavaRDDLike$class.foreachPartition(JavaRDDLike.scala:219)
	at org.apache.spark.api.java.AbstractJavaRDDLike.foreachPartition(JavaRDDLike.scala:45)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.writeRepartitionAndSortedRDDToParquet(SparkDpp.java:194)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.processRollupTree(SparkDpp.java:332)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.process(SparkDpp.java:1014)
	at org.apache.doris.load.loadv2.dpp.SparkDpp.doDpp(SparkDpp.java:1050)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.processDpp(SparkEtlJob.java:142)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.processData(SparkEtlJob.java:226)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.run(SparkEtlJob.java:233)
	at org.apache.doris.load.loadv2.etl.SparkEtlJob.main(SparkEtlJob.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.yarn.ApplicationMaster$$anon$2.run(ApplicationMaster.scala:684)
Caused by: java.lang.RuntimeException: Error while encoding: java.lang.RuntimeException: java.lang.String is not a valid external type for schema of date

```

**Expected behavior**
The String type field can be transform to DATE type  automatically.

**Screenshots**
In the function convertSrcDataframeToDstDataframe(xxx), it seem to do transform string to date, but it can not work.

image
image
image

**Desktop (please complete the following information):**
 - OS: [centos7]
 - Version [0.13.0-rc03 ]

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes

Some little fix for spark load

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #5224), and have described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

File: fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java [hunk 1]
```diff
@@ -857,18 +864,39 @@ private Object convertPartitionKey(Object srcValue, Class dstClass) throws Spark
...
         List<EtlJobConfig.EtlColumn> columnNameNeedCheckArrayList = new ArrayList<>();
         List<ColumnParser> columnParserArrayList = new ArrayList<>();
         for (EtlJobConfig.EtlColumn column : baseIndex.columns) {
-            if (!StringUtils.equalsIgnoreCase(column.columnType, "varchar") &&
+            // note(wb): there are three data source for bitmap column
+            // case 1: global dict; need't check
+            // case 2: bitmap hash function; this func is not supported in spark load now, so ignore it here
+            // case 3: origin value is a integer value; it should be checked use LongParser
+            if (StringUtils.equalsIgnoreCase(column.columnType, "bitmap")) {
+                if (dictBitmapColumnSet.contains(column.columnName)) {
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java:893]: column name tolowercase

File: fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java [hunk 2]
```diff
@@ -21,18 +21,22 @@
...
 import java.math.BigInteger;
-import java.util.Date;
 
 // Parser to validate value for different type
 public abstract class ColumnParser implements Serializable {
 
     protected static final Logger LOG = LogManager.getLogger(ColumnParser.class);
 
+    // thread safe formatter
+    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("YYYY-MM-DD");
```
[the remark this answers is not in the document]
Comment (wangbo) [reply fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java:37]: Why "yyyy-MM-DD" is suggested?
Is there any difference between them?

File: fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java (hunk 1 above)
Comment (wangbo) [reply fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java:893]: 👌

File: fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (caiconghui) [reply fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java:37]: @wangbo sorry, it should be 'yyyy-MM-dd'. you can see https://zhuanlan.zhihu.com/p/101150248

[the remark this answers is not in the document]
Comment (wangbo) [reply fe/spark-dpp/src/main/java/org/apache/doris/load/loadv2/dpp/ColumnParser.java:37]: Thanks for sharing.

**Comment by wangbo** [comment]
@morningman please review this pr

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**